lib/tbot/index.ts

```diff
@@ -0,0 +1,83 @@
+import * as fs from 'fs/promises';
+import * as path from 'path';
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as yaml from 'yaml';
+
+import * as io from '../io';
+
+export interface SharedInputs {
+  proxy: string;
+  token: string;
+  debug: boolean;
+}
+
+export function getSharedInputs(): SharedInputs {
+  const proxy = core.getInput('proxy', { required: true });
+  const token = core.getInput('token', { required: true });
+
+  return {
+    proxy,
+    token,
+    debug: core.isDebug(),
+  };
+}
+
+// See https://github.com/gravitational/teleport/blob/master/lib/tbot/config/config.go#L206
+// For configuration references
+export interface ConfigurationV1Destination {
+  directory: {
+    path: string;
+  };
+  roles: Array<string>;
+  kubernetes_cluster?: string;
+}
+export interface ConfigurationV1 {
+  auth_server: string;
+  oneshot: boolean;
+  debug: boolean;
+  onboarding: {
+    join_method: string;
+    token: string;
+  };
+  storage: {
+    memory?: boolean;
+    directory?: string;
+  };
+  destinations: Array<ConfigurationV1Destination>;
+}
+
+export function baseConfigurationFromSharedInputs(
+  inputs: SharedInputs
+): ConfigurationV1 {
+  return {
+    auth_server: inputs.proxy,
+    oneshot: true,
+    debug: inputs.debug,
+    onboarding: {
+      join_method: 'github',
+      token: inputs.token,
+    },
+    storage: {
+      // We use memory storage here so we avoid ever writing the bots more
+      // powerful credentials to disk.
+      memory: true,
```

Comment:
@timothyb89 - Given that GitHub bots are ephemeral, this feels safe. Am I missing anything here that would make the memory storage backend a poor choice ?

Reply:
I don't see why not, we exercise it in tests regularly, and given the trivial implementation there's not a huge surface area for bugs.